### PR TITLE
Fix predefined groups mapper

### PR DESCRIPTION
**Pull request by lexcao**
Closes #14516

### CODE REVIEW

File: services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocolFactory.java [hunk 1]
```diff
@@ -34,16 +34,7 @@
...
-import org.keycloak.protocol.oidc.mappers.AddressMapper;
-import org.keycloak.protocol.oidc.mappers.AllowedWebOriginsProtocolMapper;
-import org.keycloak.protocol.oidc.mappers.AudienceResolveProtocolMapper;
-import org.keycloak.protocol.oidc.mappers.FullNameMapper;
-import org.keycloak.protocol.oidc.mappers.UserAttributeMapper;
-import org.keycloak.protocol.oidc.mappers.UserClientRoleMappingMapper;
-import org.keycloak.protocol.oidc.mappers.UserPropertyMapper;
-import org.keycloak.protocol.oidc.mappers.UserRealmRoleMappingMapper;
-import org.keycloak.protocol.oidc.mappers.UserSessionNoteMapper;
+import org.keycloak.protocol.oidc.mappers.*;
```
Comment (edewit) [inline services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocolFactory.java:37]: don't do this

File: services/src/main/java/org/keycloak/protocol/oidc/mappers/GroupMembershipMapper.java [hunk 2]
```diff
@@ -104,9 +104,9 @@ protected void setClaim(IDToken token, ProtocolMapperModel mappingModel, UserSes
     }
 
     public static ProtocolMapperModel create(String name,
-                                      String tokenClaimName,
-                                      boolean consentRequired, String consentText,
-                                      boolean accessToken, boolean idToken) {
+                                             String tokenClaimName,
+                                             boolean accessToken,
+                                             boolean idToken) {
```
Comment (edewit) [inline services/src/main/java/org/keycloak/protocol/oidc/mappers/GroupMembershipMapper.java:109]: these changes seem unrelated

**Review by edewit — CHANGES_REQUESTED** [review]
could you update and rebase and add a test

File: services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocolFactory.java [hunk 3]
```diff
@@ -210,7 +211,7 @@ void initBuiltIns() {
                 true, true);
         builtins.put(UPN, model);
 
-        model = UserRealmRoleMappingMapper.create(null, GROUPS, GROUPS, true, true, true);
```
Comment (pedroigor) [inline services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocolFactory.java:213]: I'm afraid you can't change this one because this built-in mapper is used when you have MicroProfile-based clients. There, the `realm roles` (not group membership) is added to a specific `groups` claim in the token.

See https://github.com/keycloak/keycloak/pull/5933.

**Review by pedroigor — CHANGES_REQUESTED** [review]
@lexcao I can't access the issue you linked but I think I see the problem you are trying to solve.

When listing built-in mappers, you see a `groups` mapper that is actually mapping realm roles, is that correct?

It is a valid issue and we could try to remove this mapper from the built-ins as it is very specific to when you create/associate the `microprofile-jwt` to your client.

@sguilhen Wdyt?

File: services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocolFactory.java (hunk 3 above)
Comment (sguilhen) [reply services/src/main/java/org/keycloak/protocol/oidc/OIDCLoginProtocolFactory.java:213]: Yeah, in MP-JWT the roles are added to the `groups` claim.

**Comment by lexcao** [comment]
Hi @pedroigor 
I can not access the issue either. Maybe the issue was removed.

> you see a groups mapper that is actually mapping realm roles, is that correct?

Yes, it was mapped to roles not groups.

> It is a valid issue and we could try to remove this mapper from the built-ins as it is very specific to when you create/associate the microprofile-jwt to your client.

Thanks for confirming. I will create an issue to track this.

I am going to make the `groups` mapper be more clear, please help review on the this @sguilhen 
1. remove `groups` mapper from built-in since it was the `roles` mapper.
2. update the `MP-JWT` to use the `groups` to map the roles.

**Comment by pedroigor** [comment]
Thanks, @lexcao. Closing in favor of the upcoming improvement to the mapper documentation/description.

**Comment by lexcao** [comment]
Looks fine to me, thanks!
